app/models/git_repository.rb

```diff
@@ -107,4 +108,8 @@ def repo_params
     end
     params
   end
+
+  def delete_repo_dir
+    FileUtils.rm_rf(directory_name) if Dir.exist?(directory_name)
```

Comment:
`FileUtils.rm_rf` does not raise an error if the directory does not exist, so no need for the `Dir.exist?` check.